src/solvers/refinement/string_constraint_generator_concat.cpp

```diff
@@ -11,44 +11,60 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 
 /*******************************************************************\
 
-Function: string_constraint_generatort::add_axioms_for_concat
+Function: string_constraint_generatort::add_axioms_for_concat_substr
 
-  Inputs: two string expressions
+  Inputs:
+    s1 - string expression
+    s2 - string expression
+    start_index - expression representing an integer
+    end_index - expression representing an integer
 
  Outputs: a new string expression
 
  Purpose: add axioms to say that the returned string expression is equal to
-          the concatenation of the two string expressions given as input
+          the concatenation of s1 with the substring of s2 starting at index
+          start_index and ending at index end_index.
+          If start_index >= end_index the value returned is s1.
+
+          TODO: For now it just assumes end_index < s2.length. This
```

Comment:
I also want to voice my support for this: Assumptions should always be tested to see if they hold, otherwise they can lead to nasty bugs that are hard to debug.